File: python/gopigoflask.py
#!/usr/bin/env python
import atexit
import flask
import gopigo
import logging
import time

atexit.register(gopigo.stop)


app = flask.Flask(__name__)

@app.route("/")
def hello():
    return flask.render_template('index.html')

@app.route("/command/<command>")
def do_command(command=None):
    logging.debug(command)
    if command in ["forward","fwd"]:
        gopigo.fwd()
    elif command == "left":
        gopigo.left()
    elif command == "left_rot":
        gopigo.left_rot()
    elif command == "right":
        gopigo.right()
    elif command == "right_rot":
        gopigo.right_rot()
    elif command == "stop":
        gopigo.stop()
    elif command == "leftled_on":
        gopigo.led_on(0)
    elif command == "leftled_off":
        gopigo.led_off(0)
    elif command == "rightled_on":
        gopigo.led_on(1)
    elif command == "rightled_off":
        gopigo.led_off(1)
    elif command in ["back","bwd"]:
        gopigo.bwd()
    elif command == "speed":
        logging.debug("speed")
        speed = flask.request.args.get("speed")
        logging.debug("speed:" + str(speed))
        if speed:
            logging.debug("in if speed")
            gopigo.set_speed(int(speed))
        left_speed = flask.request.args.get("left_speed")
        logging.debug("left_speed:" + str(left_speed))
        if left_speed:
            logging.debug("in if left_speed")
            gopigo.set_left_speed(int(left_speed))
        right_speed = flask.request.args.get("right_speed")
        logging.debug("right_speed:" + str(right_speed))
        if right_speed:
            logging.debug("in if right_speed")
            gopigo.set_right_speed(int(right_speed))
        speed_result = gopigo.read_motor_speed()
        logging.debug(speed_result)
        return flask.json.jsonify({'speed':speed_result,'right':speed_result[0],'left':speed_result[1]})
    return ""

if __name__ == "__main__":
    logging.basicConfig(level=logging.DEBUG)
    app.run(host='0.0.0.0')

File: python/static/gopigoflask.js
function httpGet(theUrl, updateElement)
{
    if (window.XMLHttpRequest)
    {// code for IE7+, Firefox, Chrome, Opera, Safari
        xmlhttp=new XMLHttpRequest();
    }
    else
    {// code for IE6, IE5
        xmlhttp=new ActiveXObject("Microsoft.XMLHTTP");
    }
    xmlhttp.onreadystatechange=function()
    {
        if (xmlhttp.readyState==4 && xmlhttp.status==200)
        {
            console.log(xmlhttp.responseText);
            if (updateElement){
                updateElement.innerHTML = xmlhttp.responseText;
            }
        }
    }
    xmlhttp.open("GET", theUrl, true );
    xmlhttp.send();
}

var gopigo = gopigo || {};
gopigo.fwd = function(){httpGet("/command/fwd");}
gopigo.bwd = function(){httpGet("/command/bwd");}
gopigo.stop = function(){httpGet("/command/stop");}
gopigo.bwd = function(){httpGet("/command/bwd");}
gopigo.right = function(){httpGet("/command/right");}
gopigo.right_rot = function(){httpGet("/command/right_rot");}
gopigo.left = function(){httpGet("/command/left");}
gopigo.left_rot = function(){httpGet("/command/left_rot");}
gopigo.rightled_on = function(){httpGet("/command/rightled_on");}
gopigo.rightled_off = function(){httpGet("/command/rightled_off");}
gopigo.leftled_on = function(){httpGet("/command/leftled_on");}
gopigo.leftled_off = function(){httpGet("/command/leftled_off");}
gopigo.speed = function(speed, left_speed, right_speed, updateElement){
    var command = "/command/speed"
    if (speed || left_speed || right_speed){
        var command = "/command/speed?";
        var add_ampersand = null;
        if (speed){
            command = command + "speed=" + speed;
            add_ampersand = true;
        }
        if (left_speed){
            if (add_ampersand){
                command = command + "&";
            }
            command = command + "left_speed=" + left_speed;
            add_ampersand = true;
        }
        if (right_speed){
            if (add_ampersand){
                command = command + "&";
            }
            command = command + "right_speed=" + right_speed;
            add_ampersand = true;
        }
        httpGet(command, updateElement);
    }
}

